springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/AbstractOperationDataScanner.java

```diff
@@ -100,17 +100,50 @@ private Message buildMessage(OperationData operationData) {
         String modelName = this.getSchemaService().register(payloadType);
         String headerModelName = this.getSchemaService().register(operationData.getHeaders());
 
-        Schema schema = payloadType.getAnnotation(io.swagger.v3.oas.annotations.media.Schema.class);
+        /*
+         * Message information can be obtained via a @AsyncMessage annotation on the method parameter, the Payload
+         * itself or via the Swagger @Schema annotation on the Payload.
+         */
+
+        var schema = payloadType.getAnnotation(Schema.class);
         String description = schema != null ? schema.description() : null;
 
-        return Message.builder()
+        var builder = Message.builder()
                 .name(payloadType.getName())
                 .title(modelName)
                 .description(description)
                 .payload(PayloadReference.fromModelName(modelName))
                 .headers(HeaderReference.fromModelName(headerModelName))
-                .bindings(operationData.getMessageBinding())
-                .build();
+                .bindings(operationData.getMessageBinding());
+
+        // Retrieve the Message information obtained from the @AsyncMessage annotation. These values have higher priority
+        //  so if we find them, we need to override the default values.
+        processAsyncMessageAnnotation(operationData.getMessage(), builder);
+
+        return builder.build();
     }
 
+    private void processAsyncMessageAnnotation(Message annotationMessage, Message.MessageBuilder builder) {
+        if (annotationMessage != null) {
+            builder.messageId(annotationMessage.getMessageId());
+
+            var schemaFormat = annotationMessage.getSchemaFormat() != null ? annotationMessage.getSchemaFormat() : Message.DEFAULT_SCHEMA_FORMAT;
+            builder.schemaFormat(schemaFormat);
+
+            var annotationMessageDescription = annotationMessage.getDescription();
+            if (annotationMessageDescription != null && !annotationMessageDescription.isBlank()) {
+                builder.description(annotationMessageDescription);
+            }
+
+            var name = annotationMessage.getName();
+            if (name != null && !name.isBlank()) {
+                builder.name(annotationMessage.getName());
+            }
+
+            var title = annotationMessage.getTitle();
+            if (title != null && !title.isBlank()) {
```

Comment:
This can be simplified to: `StringUtils.isNoneBlank(title)`

Reply:
That would require adding a dependency to Apache Commons. Maybe using `StringUtils.hasText()` (from already imported Spring-Framework) would be enough.